modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -1377,61 +1393,89 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
 
   private def unfoldNodeAndTypeP(
       node: Node,
-      p: Primitive
-  ): TypedNode[NodeAndType] = (node, p) match {
-    // String
-    case (N.StringNode(str), Primitive.String) =>
-      TypedNode.PrimitiveTN(Primitive.String, str)
-    // Numeric
-    case (N.NumberNode(num), Primitive.Int) =>
-      TypedNode.PrimitiveTN(Primitive.Int, num.intValue())
-    case (N.NumberNode(num), Primitive.Long) =>
-      TypedNode.PrimitiveTN(Primitive.Long, num.longValue())
-    case (N.NumberNode(num), Primitive.Double) =>
-      TypedNode.PrimitiveTN(Primitive.Double, num.doubleValue())
-    case (N.NumberNode(num), Primitive.Float) =>
-      TypedNode.PrimitiveTN(Primitive.Float, num.floatValue())
-    case (N.NumberNode(num), Primitive.Short) =>
-      TypedNode.PrimitiveTN(Primitive.Short, num.shortValue())
-    case (N.NumberNode(num), Primitive.BigDecimal) =>
-      TypedNode.PrimitiveTN(Primitive.BigDecimal, BigDecimal(num.doubleValue()))
-    case (N.NumberNode(num), Primitive.BigInteger) =>
-      TypedNode.PrimitiveTN(Primitive.BigInteger, BigInt(num.intValue()))
-    // Boolean
-    case (N.BooleanNode(bool), Primitive.Bool) =>
-      TypedNode.PrimitiveTN(Primitive.Bool, bool)
-    case (node, Primitive.Document) =>
-      TypedNode.PrimitiveTN(Primitive.Document, node)
-    case (node, Primitive.String) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.String, "")
-    case (node, Primitive.Int) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Int, 0)
-    case (node, Primitive.Long) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Long, 0L)
-    case (node, Primitive.Double) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Double, 0.0)
-    case (node, Primitive.Float) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Float, 0.0f)
-    case (node, Primitive.Short) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Short, 0: Short)
-    case (node, Primitive.Byte) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Byte, 0.toByte)
-    case (node, Primitive.Blob) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Blob, Array.empty[Byte])
-    case (node, Primitive.Bool) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(Primitive.Bool, false)
-    case (node, Primitive.Timestamp) if node == Node.nullNode =>
-      TypedNode.PrimitiveTN(
-        Primitive.Timestamp,
-        java.time.Instant.ofEpochSecond(0)
+      p: Primitive,
+      traits: List[Trait]
```

Comment:
let's avoid passing the trait and just do something like : `parseEpoch(node).orElse(parseDateTime(node)).orElse(parseHttpDate(node))`

Reply:
The `node` that is passed here lacks this information, it is just the value of the `@default` trait. Don't like the `traits` being passed particularly too, but wasn't sure what is the best way to pass the `@timestampFormat` information, given that this is used in other places (and other types) too.

Reply:
My point is that it's okay to not pass the timestamp format and just try all officially supported formats in sequence. I think that timestampFormat is probably the only trait that is taken into consideration by the canonical node representation of data in smithy, and given that the data is already validated by the smithy layer, it's fine to just assume that one of the formats we know is supported will be successful when decoding the value.